pkg/backporter/backporter.go

```diff
@@ -553,30 +558,118 @@ func CreateCloneHandler(client bugzilla.Client, allTargetVersions sets.String, m
-		cloneID, err := client.CloneBug(bug)
+		if !allTargetVersions.Has(req.FormValue("release")) {
+			handleError(w, fmt.Errorf("invalid argument - %s is not a valid TargetRelease, must be one of %v", req.FormValue("release"), allTargetVersions.List()), fmt.Sprintf("invalid argument - %s is not a valid TargetRelease, must be one of %v", req.FormValue("release"), allTargetVersions.List()), http.StatusBadRequest, endpoint, bugID, m)
+			return
+		}
+
+		// Get clones and sort them
+		clones, err := client.GetAllClones(bug)
 		if err != nil {
-			handleError(w, err, "clone creation failed", http.StatusInternalServerError, endpoint, bugID, m)
+			handleError(w, err, fmt.Sprintf("unable to convert \"ID\" parameter from string to int: %s", req.FormValue("ID")), http.StatusBadRequest, req.URL.Path, 0, m)
 			return
 		}
-		targetRelease := bugzilla.BugUpdate{
-			TargetRelease: []string{
-				req.FormValue("release"),
-			},
+		sort.SliceStable(clones, func(i, j int) bool {
+			if len(clones[i].TargetRelease) == 0 && len(clones[j].TargetRelease) == 0 {
+				return false
+			} else if len(clones[i].TargetRelease) == 0 {
+				return true
+			} else if len(clones[j].TargetRelease) == 0 {
+				return false
+			}
+			return clones[i].TargetRelease[0] < clones[j].TargetRelease[0]
+		})
+		toCloneMajorMinorRelease, err := getMajorMinorRelease(req.FormValue("release"))
+		if err != nil {
+			handleError(w, err, fmt.Sprintf("%s cannot be parsed as release, must be of the form X.X.X", req.FormValue("release")), http.StatusBadRequest, endpoint, bugID, m)
+			return
+		}
+		var sourceBug *bugzilla.Bug
+		var sourceBugMajorMinorRel string
+		for _, clone := range clones {
+			if len(clone.TargetRelease) == 0 || clone.TargetRelease[0] == "---" {
+				continue
+			}
+			cloneMajorMinorRel, err := getMajorMinorRelease(clone.TargetRelease[0])
+			if err != nil {
+				handleError(w, err, fmt.Sprintf("%s cannot be parsed as release, must be of the form X.X.X", clone.TargetRelease[0]), http.StatusBadRequest, endpoint, bugID, m)
+				return
+			}
+			if toCloneMajorMinorRelease == cloneMajorMinorRel {
+				handleError(w, err, fmt.Sprintf("clone for major release %s already exists", clone.TargetRelease[0]), http.StatusBadRequest, endpoint, bugID, m)
+				return
+			}
+			if cloneMajorMinorRel > toCloneMajorMinorRelease {
+				sourceBug = clone
+				sourceBugMajorMinorRel = cloneMajorMinorRel
+				break
+			}
 		}
-		// Updating the cloned bug with the right target version
-		if err := client.UpdateBug(cloneID, targetRelease); err != nil {
-			handleError(w, err, fmt.Sprintf("clone created - Bug#%d, but failed to specify version for the cloned bug.", cloneID), http.StatusInternalServerError, endpoint, bugID, m)
+		if sourceBug == nil {
+			handleError(w, err, fmt.Sprintf("one bug with greater release needs to be present to clone from (highest release present is %s)", clones[len(clones)-1].TargetRelease[0]), http.StatusBadRequest, endpoint, bugID, m)
```

Comment:
Shouldn't the behavior be to backfill clones for the missing ones? So if I have one for 4.7 and want to backport to 4.3 the tool will create 4.6, 4.5, 4.4 ?

Reply:
Yes, that would be the expected behavior.
This error is expected when the user tries to create a clone with 4.6.0 as the TargetRelease even though the latest targetRelease that already exists is 4.5.0